docs(hub): what's new; hub-related announcements (#9478)
camunda/camunda-docs: docs/reference/announcements-release-notes/8100/whats-new-in-810.md

:::

## Web Modeler data

On 29 August 2026, your Web Modeler data will receive three updates:

- **[Organizational structure](#organizational-structure):** Enforces a stricter, more scalable file resource hierarchy.
- **[Data migration](#migration):** Aligns your existing data with the new structure.
- **[Process application versioning model](#process-application-versioning):** Provides more granular control.

### Organizational structure

> In Camunda 8.9, a project can contain process applications, folders, and loose files. Camunda 8.10 enforces a new file resource hierarchy in which projects only contain process applications. Everything else, including files, folders, and IDP application content, lives inside a process application.

Reviewer: ```suggestion
In Camunda 8.9, a project can contain process applications, folders, and files. Camunda 8.10 introduces a new file resource hierarchy in which projects only contain process applications. Everything else, including files, folders, and IDP application content, is stored inside a process application.
```
Few wording suggestions

For example, if this is what your data looks like in 8.9:

```txt title="Camunda 8.9"
Payments (Project)
├─ main.bpmn
├─ eligibility.dmn
├─ readme.md
├─ Forms (Folder)
│   ├── details.form
│   └── review.form
├─ Archive (Folder)
│   └── Refunds (Process application)
│       ├── refunds.bpmn
│       └── refund-request.form
└─ Onboarding (Process application)
    ├── onboarding.bpmn
    └── kyc-checks.dmn
```

This is what the data looks like in 8.10:

```txt title="Camunda 8.10"
Payments (Project)
├─ Payments - General (Process application - TEMPORARY PLACEMENT)
│   ├─ main.bpmn
│   ├─ eligibility.dmn
│   ├─ readme.md
│   ├─ Forms (Folder)
│   │   ├── details.form
│   │   └── review.form
│   └─ Archive (Folder)
├── Refunds (Process application - MOVED)
│   ├── refunds.bpmn
│   └── refund-request.form
└─ Onboarding (Process application)
    ├── onboarding.bpmn
    └── kyc-checks.dmn
```

> With a strict **Project > Process application > File/folder** hierarchy, resources are more discoverable and projects more scalable.

Reviewer: ```suggestion
This strict new **Project > Process application > File/folder** hierarchy makes resources more discoverable and your projects more scalable.
```

> ### Migration

Reviewer: Should you name this Data migration as per the heading above, for consistency? Same with versioning heading/link.

> For Self-Managed, your data will be migrated to the new organizational structure when you upgrade to Camunda 8.10. For SaaS, this happens automatically during a scheduled maintenance window.

Reviewer: ```suggestion
For Self-Managed, your data will be migrated to the new organizational structure during upgrade to Camunda 8.10. For SaaS, this happens automatically during a scheduled maintenance window.
```
Not sure if this is right - but does it happen after (implied by when) or as part of the upgrade (during)?

Author: During/as part of

> In the migration:

Reviewer: ```suggestion
During the migration:
```
Suggestion - could be wrong

- Any process application nested inside a folder moves to the top level of its project.

> - Any files or folders sitting directly in a project, not inside a process application, are automatically grouped in a new process application, named `YOUR PROJECT NAME - General`. When the migration is complete, you can rename this application, move content out of it, or otherwise reorganize it like any other process application.

Reviewer: ```suggestion
- Any files or folders located directly in a project, not inside a process application, are automatically grouped in a new process application, named `YOUR PROJECT NAME - General`. Once the migration is complete, you can rename this application, move content out of it, or reorganize it as with any other process application.
```

> - Git sync and cluster settings on existing process applications migrate along with your data. They carry over untouched.

Reviewer: ```suggestion
- Git sync and cluster settings on existing process applications migrate unchanged along with your data.
```

> During the migration, Web Modeler is briefly unavailable. Clusters and running processes are unaffected and keep executing normally.

Reviewer: ```suggestion
During the migration, Web Modeler is briefly unavailable. Clusters and running processes are unaffected and continue executing normally.
```

| Running process instances           | Orchestration Clusters, engines, and running process instances continue unaffected. Web Modeler and Camunda Hub form a modeling-and-management plane independent of the runtime path.     |
| Redeployment                        | Deployments live on your clusters, not in the project structure being reorganized. Everything already deployed keeps running; this change doesn't require redeployment.                   |
| Clusters and configuration          | Cluster and deployment settings attached to existing process applications migrate along with the data and carry over unchanged.                                                           |
| Files, folders, and version history | All files, folders, versions, and history are preserved; only their location within the project changes.                                                                                  |
| Git-synced projects                 | The migration doesn't modify process applications or their contents, so anything connected through Git sync is unaffected: the same files, in the same repository, with the same history. |

> | Desktop Modeler                     | Desktop Modeler has no direct connection to Web Modeler that this change touches, and content shared via Git sync is unaffected.                                                          |

Reviewer: ```suggestion
| Desktop Modeler                     | Desktop Modeler has no direct connection to Web Modeler affected by this change. Content shared via Git sync is also unaffected.                                                          |
```

> If you automate against the Web Modeler API, content moving to a new location affects you directly: the V1 API returns files and folders from their new place, and requests that create an item at a project's root are redirected into the new `YOUR PROJECT NAME - General` application, with the response reflecting the new location.

Reviewer: An example could be useful here, but not essential

Author: I'll merge for now and add an example if I have time later

> Review any automation that depends on where files or folders sit before the migration runs. A small number of folder API integrations are affected more directly; if you rely on the folder API to work with process applications, contact support to confirm whether your integration needs updates.

Reviewer: ```suggestion
You should review any automation that relies on where files or folders are located before running the migration. A small number of folder API integrations are affected more directly; if you rely on the folder API to work with process applications, contact support to confirm whether your integration needs updates.
```

Author: > ...before running the migration.

While I also prefer the active voice in cases like this, I don't think this is 100% accurate here. SaaS customers won't run the migration themselves, we will do it on their behalf, so I'll leave that part alone until we update this content to cater more to SM users.

### Process application versioning

> In addition to the Web Modeler data migration, 8.10 brings an improved versioning model for process applications:

Reviewer: ```suggestion
In addition to the Web Modeler data migration, 8.10 provides an improved process application versioning model:
```

- File-level versions — process applications can be versioned as a bundle, as before, but now also at the single-file level.
- Auto-save for all files, plus file-level version history for every file.
- Decoupled versioning — process application versions and element template versions are now created independently of each other.

> A process application is no longer a rigid bundle. You can use it as a bundle of assets deployed together and Git-synced, or as a plain container for loose files and folders--whichever fits the work. This flexibility is why everything now lives inside a process application: being in one doesn't lock you into bundle behavior.

Reviewer: ```suggestion
A process application is no longer a rigid bundle. You can use it as a bundle of assets deployed together and Git-synced, or as a simple container for loose files and folders. This flexibility is why everything now lives inside a process application: being in one doesn't lock you into bundle behavior.
```
I'm sure this could be reworded or made more concise, but not sure what that would be - the last sentence seems clunky?

Author: I agree. I missed this sentence when I carried it over. I'll rewrite it.
